core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -131,7 +131,10 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
     lazy val cachedSerializer = SparkEnv.get.serializer.newInstance()
     val createZero = () => cachedSerializer.deserialize[U](ByteBuffer.wrap(zeroArray))
 
-    combineByKey[U]((v: V) => seqOp(createZero(), v), seqOp, combOp, partitioner)
+    val cleanedSeqOp = self.context.clean(seqOp)
+    val cleanedCombOp = self.context.clean(combOp)
+    combineByKey[U](
+      (v: V) => cleanedSeqOp(createZero(), v), cleanedSeqOp, cleanedCombOp, partitioner)
```

Comment:
I think we have a rule here implicitly. All public methods in RDD(PairRDDFunctions, ...) that accepts closure arguments should be responsible for cleaning the closures. 

So we don't need to track where the closures end up. Just check if the closure argument is passed to a public method. If so, we don't need to clean it. Otherwise, it should be cleaned.